src/common/FileSystem.cpp

```diff
@@ -2155,37 +2171,77 @@ bool IsInitialized()
 // Add a pak to the list of available paks
 static void AddPak(pakType_t type, Str::StringRef filename, Str::StringRef basePath)
 {
+	size_t suffixLen;
+	bool isPakDir = (type == pakType_t::PAK_DIR);
 	std::string fullPath = Path::Build(basePath, filename);
 
-	size_t suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
+	suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
+
 	std::string name, version;
 	Util::optional<uint32_t> checksum;
-	if (!ParsePakName(filename.begin(), filename.end() - suffixLen, name, version, checksum) || (type == pakType_t::PAK_DIR && checksum)) {
-		if (!UseLegacyPaks()) {
-			fsLogs.Warn("Invalid pak name: %s", fullPath);
+
+	if (!ParsePakName(filename.begin(), filename.end() - suffixLen, name, version, checksum) || (isPakDir && checksum)) {
+       fsLogs.Warn("Invalid pak name: '%s'", fullPath);
+       return;
+	}
+
+	if (type == pakType_t::PAK_ZIP) {
+		if (!isdigit(version.at(0))) {
+			fsLogs.Warn("Invalid pak version, must start with a digit: '%s'", fullPath);
 			return;
-		} else {
-			fsLogs.Notice("Loading legacy pak: %s", fullPath);
-			name = filename.substr(0, filename.size() - suffixLen);
-			version = "1";
 		}
+		fsLogs.Verbose("Adding pak: '%s'...", fullPath);
```

Comment:
They do not have the same purpose and are not printed for the same events. This one is printed when a pak is added to the known list of paks. It will be printed on all pak successfully found. The other one _("loading pak: ")_ is printed only when the pak is effectively loaded. The idea of this other log line is to allow user to check if his pak is found even if it's not loaded, using `+set logs.logLevel.fs debug`.